tests/test_get_spends_for_block.py

```diff
@@ -0,0 +1,69 @@
coin_info: 49d599dbb3579ac0ca5af399cce99af5f47b8263539c3219f1ef367eed472049, puzzle_hash: bbb2072b374b5b25307fbb1e11a99740d12645db6530f5249f63bf17834d77b6, amount: 0 }, puzzle_reveal: Program(ff02ffff01ff02ffff01ff02ffff03ffff15ff820bffff8080ffff01ff04ffff04ff0affff04ffff02ff0effff04ff02ffff04ffff04ff8202ffffff04ff8205ffffff04ff820bffff80808080ff80808080ff808080ffff04ffff04ff04ffff04ff05ffff04ff820bffff80808080ff808080ffff01ff08ffff019674616b65725f616d6f756e74206973206c65737320308080ff0180ffff04ffff01ff33ff3eff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff0effff04ff02ffff04ff09ff80808080ffff02ff0effff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080ffff04ffff01a002f3ff79c563e256f91cc69757eab80eb75bb86f986981ae7f35c427b0dad671ffff04ffff0105ffff04ffff01822710ffff04ffff0184616145efffff04ffff0180ffff04ffff01a02b94c947bd39ee56530bbe67e54f8feb8ac46b570d84a03b12c576225f13d53bffff04ffff01b08863944c9b54e2de3a6d3cb51d03d23b8a011be4259782b6e4cda2bdd625d8b2f489d03755336eaf8730d880f9360ab3ff018080808080808080), solution: Program(ff05ff05ff82271080) }, 'conditions': [(62, [b'\\xde\\xcf\\\\eI\\xa6S\\xa0\\xcb\\x1c\\xddSPP\\xb1o\\xc6I\\x94AjEh7\\xde\\x91\\xd3\\xc1\\xa0:7@']), (51, [b\"\\x02\\xf3\\xffy\\xc5c\\xe2V\\xf9\\x1c\\xc6\\x97W\\xea\\xb8\\x0e\\xb7[\\xb8o\\x98i\\x81\\xae\\x7f5\\xc4'\\xb0\\xda\\xd6q\", b\"'\\x10\"])]}, {'coin_spend': CoinSpend { coin: Coin { parent_coin_info: 10dccb906131a1e5be77261232d09db0b5719b6b38e3e6a825a906d465602fe7, puzzle_hash: caf0226aaaff7fec3cc55d300930f7f63b12135b9cf7c4cf8a5c431db43b3a91, amount: 3310758457 }, puzzle_reveal: Program(ff02ffff01ff02ffff01ff02ffff03ff0bffff01ff02ffff03ffff09ff05ffff1dff0bffff1effff0bff0bffff02ff06ffff04ff02ffff04ff17ff8080808080808080ffff01ff02ff17ff2f80ffff01ff088080ff0180ffff01ff04ffff04ff04ffff04ff05ffff04ffff02ff06ffff04ff02ffff04ff17ff80808080ff80808080ffff02ff17ff2f808080ff0180ffff04ffff01ff32ff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff06ffff04ff02ffff04ff09ff80808080ffff02ff06ffff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080ffff04ffff01b0a137bfa49e84c734e0ca689dbf186b98380c366ecd74011eee6840ff88b04476b327a2707852b83103e4faac53fbeb22ff018080), solution: Program(ff80ffff01ffff33ffa0d767c90667a593a8c585cd98ddf766a9d500168d84c01ad9b0dcdc349b4bbc0dff8500c556032980ffff3cffa06fb22bd61d8746104fd0a7168664b8a35c39d461f90ea4da354ba9519c43bd3f8080ff8080) }, 'conditions': [(50, [b'\\xa17\\xbf\\xa4\\x9e\\x84\\xc74\\xe0\\xcah\\x9d\\xbf\\x18k\\x988\\x0c6n\\xcdt\\x01\\x1e\\xeeh@\\xff\\x88\\xb0Dv\\xb3\\'\\xa2pxR\\xb81\\x03\\xe4\\xfa\\xacS\\xfb\\xeb\"', b'=\\x82{g\\xfco\\x90\\xb9\\xc0M\\xe7\\xcc\\xedc)\\x07T\\x8dY\\xec\\x05V\\n3\\xf3\\xcd\\x81\\xbf\\xe0\\xd8\\xbb\\xde']), (51, [b'\\xd7g\\xc9\\x06g\\xa5\\x93\\xa8\\xc5\\x85\\xcd\\x98\\xdd\\xf7f\\xa9\\xd5\\x00\\x16\\x8d\\x84\\xc0\\x1a\\xd9\\xb0\\xdc\\xdc4\\x9bK\\xbc\\r', b'\\x00\\xc5V\\x03)']), (60, [b'o\\xb2+\\xd6\\x1d\\x87F\\x10O\\xd0\\xa7\\x16\\x86d\\xb8\\xa3\\\\9\\xd4a\\xf9\\x0e\\xa4\\xda5K\\xa9Q\\x9cC\\xbd?'])]}]"
```

Comment:
this is a regression test, to fail if the output changes. It doesn't really prove that this output is *correct*, or consistent with `run_block_generator2()` (which I think it's fair to treat as the aithority on what's expected).

I think it would be good to parameterize a test on the test-generators and actually use the expected output embedded in those test cases to ensure we get something out that matches. We have tests in `crates/chia-consensus/src/test_generators.rs` which parameterize on these test cases, but it's rust and would only be feasible for the rust function.